src/main/domain/consent/LoadConsentService.js

```diff
@@ -2,25 +2,118 @@ import {inject} from '../../core/ioc/ioc'
 import {ConsentRepository} from '../consent/ConsentRepository'
 import {ConsentFactory} from '../consent/ConsentFactory'
 import {ConsentDecoderService} from '../consent/ConsentDecoderService'
+import {VendorListRepository} from '../vendorlist/VendorListRepository'
+import {ConsentEncoderService} from './ConsentEncoderService'
+import {VendorListHelper} from '../vendorlist/VendorListHelper'
 
 export class LoadConsentService {
+  _consentEncoderService
+  _vendorListHelper
   constructor({
     consentRepository = inject(ConsentRepository),
     consentFactory = inject(ConsentFactory),
-    consentDecoderService = inject(ConsentDecoderService)
+    consentDecoderService = inject(ConsentDecoderService),
+    consentEncoderService = inject(ConsentEncoderService),
+    vendorListRepository = inject(VendorListRepository),
+    vendorListHelper = inject(VendorListHelper)
   } = {}) {
     this._consentRepository = consentRepository
     this._consentFactory = consentFactory
     this._consentDecoderService = consentDecoderService
+    this._consentEncoderService = consentEncoderService
+    this._vendorListRepository = vendorListRepository
+    this._vendorListHelper = vendorListHelper
   }
 
   async loadConsent() {
     const encodedConsent = this._consentRepository.loadUserConsent()
+
     if (!encodedConsent) {
       return this._consentFactory.createEmptyConsent()
     }
-    return this._consentFactory.createConsent(
-      this._consentDecoderService.decode({encodedConsent})
+    const existingConsent = this._consentDecoderService.decode({encodedConsent})
+
+    const newVendorList = await this._vendorListRepository.getVendorList()
+
+    const result = await this._isValidAndSaveNewConsent({
+      newVendorList,
+      consent: existingConsent
+    })
+
+    return result
+      ? this._createValidConsentWithTheSavedOne()
+      : this._createAnInValidConsentAndMergeListVendors({
+          existingConsent,
+          newVendorList
+        })
+  }
+
+  _createAnInValidConsentAndMergeListVendors({existingConsent, newVendorList}) {
+    existingConsent.valid = false
+    existingConsent.vendor = this._vendorListHelper.mergeVendors({
+      newVendorList: newVendorList.vendors,
+      oldVendors: existingConsent.vendor
+    })
+    return this._consentFactory.createConsent(existingConsent)
+  }
+
+  async _createValidConsentWithTheSavedOne() {
+    const validEncodedConsent = await this._consentRepository.loadUserConsent()
+    const validConsent = this._consentDecoderService.decode({
```

Comment:
Is it necessary to get the recently stored consent (in _consentHaveConsentsAndLIForAllVendorsWithTheSameValue method) from the consent repository? Or we could get the one that that method is working with?

Reply:
I have preferred return, the saved one. For performance reasons it is better return the one that method is working with, but performance issues I think is better to attach later.